src/vm/object.h

```diff
@@ -1205,6 +1207,32 @@ class ReflectClassBaseObject : public BaseObjectWithCachedData
 
 };
 
+#include <pshpack4.h>
+class Utf8StringObject : public Object
+{
+#ifdef DACCESS_COMPILE
+    friend class ClrDataAccess;
+#endif
+
+private:
+    DWORD   m_StringLength;
+    BYTE    m_Characters[0];
```

Comment:
For the String over-allocation fix, I have changed this to `m_FirstChar` to make it look like the managed side, and not depend on pragmas that control packing and non-standard extensions.